src/components/TokenHistory.js

```diff
@@ -107,7 +117,8 @@ class TokenHistory extends React.Component {
    * Calculates the index of the reference hash in the history list
    */
   getReferenceIndex = () => {
-    const idxReference = this.props.history.findIndex((tx) =>
+    const history = this.props.tokensHistory[this.props.selectedToken];
+    const idxReference = history.findIndex((tx) =>
```

Comment:
I added this validation but actually I am throwing an error because this method should never be called when `this.state.reference = null`. Every place we call it in the code we are validating this before. What do you think?

Reply:
The way the pagination is made, throwing an error on `this.state.reference = null` is the right thing.
I think the pagination could be simpler, but it's not something for this issue.